test/mpi/datatype/hvecblklen.c

```diff
@@ -41,7 +41,7 @@ int main(int argc, char *argv[])
     outbuf = (char *) malloc(outsize);
     for (i = 0; i < outsize; i++) {
         inbuf[i] = i % 64;
-        outbuf[i] = -1;
+        outbuf[i] = (char) -1;
```

Comment:
Do we want to use `CHAR_MAX` or `CHAR_MIN` instead of a cast? Without context the cast looks a bit strange. I'm OK either way. Just a suggestion.